src/storage/phonesContext.tsx

```diff
@@ -0,0 +1,245 @@
+  cartProducts: CartProduct[],
+  setCartProducts: (v: CartProduct[]) => void,
+  handleOnLikeClick: (v: string) => void,
+  handleOnCartAdd: (v: string) => void,
+  removeCartItem: (v: string) => void,
+  plusCartItem: (v: string) => void,
+  minusCartItem: (v: string) => void,
+  getProductCount: (v: string) => number,
+};
+
+export const PhonesContext = React.createContext<PhonesContextType>({
+  products: [],
+  setProducts: () => { },
+  preparedHotPriceProducts: [],
+  preparedBrandNewProducts: [],
+  sortType: SortType.Newest,
+  setSortType: () => { },
+  itemsPerPage: 0,
+  setItemsPerPage: () => { },
+  sortParams: [],
+  perPageParams: [],
+  sortedProducts: () => [],
+  tabletSearchValue: '',
+  setTabletSearchValue: () => { },
+  phoneSearchValue: '',
+  setPhoneSearchValue: () => { },
+  filteredProducts: [],
+  suggestedProducts: [],
+  setSuggestedProducts: () => { },
+  favoritesId: [],
+  setFavoritesId: () => { },
+  cartProducts: [],
+  setCartProducts: () => { },
+  handleOnLikeClick: () => { },
+  handleOnCartAdd: () => { },
+  removeCartItem: () => { },
+  plusCartItem: () => {},
+  minusCartItem: () => {},
+  getProductCount: () => 0,
+});
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const PhonesProvider: React.FC<Props> = ({ children }) => {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.Newest);
+  const [itemsPerPage, setItemsPerPage] = useState(16);
+  const [tabletSearchValue, setTabletSearchValue] = useState('');
+  const [phoneSearchValue, setPhoneSearchValue] = useState('');
+  const [suggestedProducts, setSuggestedProducts] = useState<Product[]>([]);
+  const [
+    favoritesId, setFavoritesId,
+  ] = useLocalStorage<string[]>('favorites', []);
+  const [
+    cartProducts, setCartProducts,
+  ] = useLocalStorage<CartProduct[]>('cart', []);
+
+  const sortParams = [
+    {
+      type: SortType.Newest,
+      value: 'age',
+    },
+    {
+      type: SortType.Alphabetically,
+      value: 'name',
+    },
+    {
+      type: SortType.Cheapest,
+      value: 'price',
+    },
+  ];
+
+  const perPageParams = [4, 8, 16, 32];
+
+  const preparedHotPriceProducts = products?.filter((product: Product) => (
+    product.fullPrice - product.price >= 90
+  )) || [];
+
+  const preparedBrandNewProducts = products?.filter((product: Product) => (
+    product.year >= 2019
+  )) || [];
+
+  const sortedProducts = () => {
+    switch (sortType as SortType) {
+      case SortType.Alphabetically:
+        return [...products].sort((prev, next) => (
+          next.name.localeCompare(prev.name)
+        ));
+
+      case SortType.Cheapest:
+        return [...products]
+          .sort((prev, next) => prev.fullPrice - next.fullPrice);
+
+      default:
+        return [...products].sort((prev, next) => next.year - prev.year);
+    }
+  };
+
+  const filteredProducts = sortedProducts().filter(product => (
```

Comment:
you have all sort and filter params available on every page. So it's better to recalculate it on the page where you use it (i.e. `PhonesPage`) using params

When it is in context you recalculate it on every page when any state value changes